src/tiledb/cloud/invites.py

```diff
@@ -0,0 +1,165 @@
+
+
+def fetch_invitations(**filters):
+    """
+    Fetches a paginated list of invitations.
+
+    :param str organization: name or ID of organization to filter
+    :param str array: name/uri of array that is url-encoded to filter
+    :param str group: name or ID of group to filter
+    :param int start: start time for tasks to filter by
+    :param int end: end time for tasks to filter by
+    :param int page: pagination offset
+    :param int per_page: pagination limit
+    :param str type: invitation type, \"ARRAY_SHARE\", \"JOIN_ORGANIZATION\"
+    :param str status: Filter to only return \"PENDING\", \"ACCEPTED\"
+    :param str orderby: sort by which field valid values include
+                        timestamp, array_name, organization_name
+    :return dict: Invitations and pagination metadata.
+    """
+    invitation_api = client.build(rest_api.InvitationApi)
+    try:
+        return invitation_api.fetch_invitations(**filters)
+    except rest_api.ApiException as exc:
+        raise tiledb_cloud_error.check_exc(exc)
+
+
+def invite_to_organization(
+    organization: str, *, recipients: List[str], role: OrganizationRoles
+) -> None:
+    """
+    Sends email to multiple recipients with joining information
+    regarding an organization.
+
+    :param str organization: name or UUID of organization.
+    :param List[str] recipients: list of recipient emails
+    :param OrganizationRoles role: role assigned to the recipient.
+    :return: None
+    """
+    invitation_api = client.build(rest_api.InvitationApi)
+    email_invite = InvitationOrganizationJoinEmail(
+        organization_role=role, invitee_email=recipients
+    )
+    try:
+        return invitation_api.join_organization(organization, email_invite)
+    except rest_api.ApiException as exc:
+        raise tiledb_cloud_error.check_exc(exc)
+
+
+def cancel_invite_to_organization(*, invitation_id: str, organization: str) -> None:
+    """
+    Cancels join organization invitation.
+
+    :param str invitation_id: the ID of invitation about to be canceled.
+    :param str organization: name or UUID of organization.
+    :return: None
+    """
+    invitation_api = client.build(rest_api.InvitationApi)
+    try:
+        return invitation_api.cancel_join_organization(invitation_id, organization)
+    except rest_api.ApiException as exc:
+        raise tiledb_cloud_error.check_exc(exc)
+
+
+def invite_to_array(
+    array: str, *, recipients: List[str], role: OrganizationRoles
+) -> None:
+    """
+    Share array by email invite.
+
+    :param str array: name/uri of array that is url-encoded
+    :param List[str] recipients: list of recipient emails
+    :param OrganizationRoles role: role assigned to the recipient.
+    :return: None
+    """
+    invitation_api = client.build(rest_api.InvitationApi)
+    namespace, array = utils.split_uri(array)
+    email_invite = InvitationOrganizationJoinEmail(
+        organization_role=role, invitee_email=recipients
+    )
```

Comment:
I think you got a bit too excited with copy/paste here…

Reply:
Not really, although `InvitationOrganizationJoinEmail`'s `__init__` seems to suggest that `organizations_role` is optional, I am getting the following error when omitting it: `ValueError: Invalid value for 'organization_role', must not be 'None'`

Reply:
(this function is for inviting to an array)

Reply:
yeap, I knew I should be more caffeinated!!